ledger-tool/src/run.rs

```diff
@@ -0,0 +1,577 @@
+                if let Some(account) = bank.get_account_with_fixed_root(&account_info.key) {
+                    let owner = *account.owner();
+                    if bpf_loader_upgradeable::check_id(&owner) {
+                        if let Ok(UpgradeableLoaderState::Program {
+                            programdata_address,
+                        }) = account.state()
+                        {
+                            debug!("Program data address {}", programdata_address);
+                            if bank
+                                .get_account_with_fixed_root(&programdata_address)
+                                .is_some()
+                            {
+                                cached_account_keys.push(account_info.key);
+                            }
+                        }
+                    }
+                    transaction_accounts.push((account_info.key, account));
+                    instruction_accounts.push(InstructionAccount {
+                        index_in_transaction: index as IndexOfAccount,
+                        index_in_caller: index as IndexOfAccount,
+                        index_in_callee: index as IndexOfAccount,
+                        is_signer: account_info.is_signer.unwrap_or(false),
+                        is_writable: account_info.is_writable.unwrap_or(false),
+                    });
+                } else {
+                    let data = account_info.data.unwrap_or(vec![]);
+                    let space = data.len();
+                    let owner = account_info.owner.unwrap_or(Pubkey::new_unique());
+                    let lamports = account_info.lamports.unwrap_or(0);
+                    let mut account = AccountSharedData::new(lamports, space, &owner);
+                    account.set_data(data);
+                    transaction_accounts.push((account_info.key, account));
+                    instruction_accounts.push(InstructionAccount {
+                        index_in_transaction: index as IndexOfAccount,
+                        index_in_caller: index as IndexOfAccount,
+                        index_in_callee: index as IndexOfAccount,
+                        is_signer: account_info.is_signer.unwrap_or(false),
+                        is_writable: account_info.is_writable.unwrap_or(false),
+                    });
+                }
+            }
+            input.instruction_data
+        }
+    };
+    let program_index: u16 = instruction_accounts.len().try_into().unwrap();
+    transaction_accounts.push((
+        loader_id,
+        AccountSharedData::new(0, 0, &solana_sdk::native_loader::id()),
+    ));
+    transaction_accounts.push((
+        program_id, // ID of the loaded program. It can modify accounts with the same owner key
+        AccountSharedData::new(0, 0, &loader_id),
+    ));
+    let interpreted = matches.value_of("use").unwrap() != "jit";
+    let ta = transaction_accounts.clone();
+    let builtin_programs = BuiltinPrograms {
```

Comment:
Why not use the [built-in programs from the Bank](https://github.com/solana-labs/solana/blob/7e2b1a885f92a7d0f6758bb417aa5d38ace7abe1/runtime/src/bank.rs#L993)?
You can make it public or add a get method.
Otherwise it can only use one of the 7 default built-ins.

Reply:
Good point. Will change to use Bank builtin programs.

Reply:
done